src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.UserDao;
+import com.internet.shop.exception.DataProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String query = "SELECT * FROM users WHERE deleted = FALSE AND login = ?";
+        User user = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                user = extractUser(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't extract user with login: " + login, e);
+        }
+        return getUserWithRoles(user);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (name, login, password) VALUES (?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+                addUserRoles(user.getId(), user.getRoles());
```

Comment:
Due to point 10 in checklist it is a bad practice to open new connection why old is not closed

Reply:
\+